src/shell/shell_cmds.cpp

```diff
@@ -496,6 +497,49 @@ static std::vector<int> map_name_lengths(const std::vector<DtaResult> &results,
 	return ws;
 }
 
+static std::vector<int> calc_column_widths(const std::vector<int> &cell_widths,
+                                           int min_cell_width)
+{
+	assert(min_cell_width > 0);
+
+	// Actual text columns using current text mode;
+	// in practice it's either 40, 80, or 132.
+	const int tcols = real_readw(BIOSMEM_SEG, BIOSMEM_NB_COLS);
+
+	// Use tcols-1 because we never want to print line up to the actual
+	// limit; this would cause unnecessary line wrapping
+	const size_t max_columns = (tcols - 1) / min_cell_width;
+	std::vector<int> col_widths(max_columns);
```

Comment:
Is `cell_width` actually the column width, in characters?

The calculation would be more intuitive with names like:

`max_columns = screen_width / min_col_width; // where width is in characters.`